packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -218,3 +218,75 @@ export function useSupportedStyles( name, element ) {
 
 	return supportedPanels;
 }
+
+/**
+ * Given a settings object and a list of supported panels,
+ * returns a new settings object with the unsupported panels removed.
+ *
+ * @param {Object}   settings Settings object.
+ * @param {string[]} supports Supported style panels.
+ *
+ * @return {Object} Merge of settings and supports.
+ */
+export function overrideSettingsWithSupports( settings, supports ) {
```

Comment:
The idea is that ultimately this function is going to allow us to merge "block.json" supports with "theme.json" settings "like" object.